### PR TITLE
Pool and Pool stable: adds check if the swapped token is actualy one of the two tokens in the liquidity pool.

### PR DESCRIPTION
**Pull request by gangov**

### CODE REVIEW

**Comment by gangov** [comment]
the issue is present also in pool stable

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,8 +16,10 @@ and this project adheres to
 ## Added
 
 - Adds a new macro that validates the bps arguments value ([#199])
+- Pool and Pool stable: adds check if the swapped token is actualy one of the two tokens in the liquidity pool ([#237])
```
Comment (ueco-jb) [inline CHANGELOG.md:19]:
```suggestion
- Pool and Pool stable: extend check if the swapped token is within pool ([#237])
```

File: contracts/pool/src/contract.rs [hunk 2]
```diff
@@ -568,6 +568,10 @@ impl LiquidityPoolTrait for LiquidityPool {
     fn simulate_swap(env: Env, offer_asset: Address, offer_amount: i128) -> SimulateSwapResponse {
         let config = get_config(&env);
 
+        if offer_asset != config.token_a && offer_asset != config.token_b {
+            panic!("Trying to swap wrong asset. Aborting..")
```
Comment (ueco-jb) [inline contracts/pool/src/contract.rs:572]: Use `soroban_sdk::panic_with_error`.

File: contracts/pool/src/tests/swap.rs [hunk 3]
```diff
@@ -724,3 +724,58 @@ fn test_swap_fee_variants(swap_fees: i64, commission_fee: i128) {
...
+    token1.mint(&user1, &1_001_000);
+    token2.mint(&user1, &1_001_000);
+    pool.provide_liquidity(
+        &user1,
+        &Some(1_000_000),
+        &Some(1_000_000),
+        &Some(1_000_000),
+        &Some(1_000_000),
+        &None,
+    );
```
Comment (ueco-jb) [inline contracts/pool/src/tests/swap.rs:767]:
```suggestion
```

File: contracts/pool/src/contract.rs [hunk 4]
```diff
@@ -603,6 +607,10 @@ impl LiquidityPoolTrait for LiquidityPool {
     ) -> SimulateReverseSwapResponse {
         let config = get_config(&env);
 
+        if ask_asset != config.token_a && ask_asset != config.token_b {
+            panic!("Trying to swap wrong asset. Aborting..")
+        }
+
```
Comment (ueco-jb) [inline contracts/pool/src/contract.rs:613]: This is superflous. Literally 4 lines belowe we do it anyway. Modify it:
```
        let (pool_balance_offer, pool_balance_ask) = if ask_asset == config.token_b {
            (pool_balance_a, pool_balance_b)
        } else if ask_asset == config.token_a {
            (pool_balance_b, pool_balance_a)
        } else {
            panic_with_error!
        };
```
Repeat in all places.

File: contracts/pool_stable/src/contracterror.rs
Comment (ueco-jb) [inline contracts/pool_stable/src/contracterror.rs:1]: Rename to `error.rs`

**Comment by gangov** [comment]
> You added one test for swap, what's with all the other checks?

added more tests, the only place that I didn't directly test is `split_deposit_based_on_pool_ratio`, but that's handled by the tests for `do_swap()`

File: contracts/pool/src/contract.rs [hunk 5]
```diff
@@ -582,8 +582,10 @@ impl LiquidityPoolTrait for LiquidityPool {
         let pool_balance_b = utils::get_pool_balance_b(&env);
         let (pool_balance_offer, pool_balance_ask) = if offer_asset == config.token_a {
             (pool_balance_a, pool_balance_b)
-        } else {
+        } else if offer_asset == config.token_b {
             (pool_balance_b, pool_balance_a)
+        } else {
+            panic_with_error!(env, ContractError::AssetNotInPool);
```
Comment (ueco-jb) [inline contracts/pool/src/contract.rs:588]: Add log with the proper message.

**Review by ueco-jb — APPROVED** [review]
LGTM

**Pull request merged**